test/utils/cryptography/ECDSA.test.js

```diff
@@ -1,104 +1,101 @@
-require('@openzeppelin/test-helpers');
-const { expectRevertCustomError } = require('../../helpers/customError');
-const { toEthSignedMessageHash } = require('../../helpers/sign');
-
+const { ethers } = require('hardhat');
 const { expect } = require('chai');
+const { loadFixture } = require('@nomicfoundation/hardhat-network-helpers');
 
-const ECDSA = artifacts.require('$ECDSA');
-
-const TEST_MESSAGE = web3.utils.sha3('OpenZeppelin');
-const WRONG_MESSAGE = web3.utils.sha3('Nope');
-const NON_HASH_MESSAGE = '0x' + Buffer.from('abcd').toString('hex');
+const TEST_MESSAGE = ethers.id('OpenZeppelin');
+const WRONG_MESSAGE = ethers.id('Nope');
+const NON_HASH_MESSAGE = '0xabcd';
 
-function to2098Format(signature) {
-  const long = web3.utils.hexToBytes(signature);
-  if (long.length !== 65) {
-    throw new Error('invalid signature length (expected long format)');
-  }
-  if (long[32] >> 7 === 1) {
-    throw new Error("invalid signature 's' value");
-  }
-  const short = long.slice(0, 64);
-  short[32] |= long[64] % 27 << 7; // set the first bit of the 32nd byte to the v parity bit
-  return web3.utils.bytesToHex(short);
+function toERC2098Format(signature) {
+  return ethers.Signature.from(signature).compactSerialized;
 }
 
 function split(signature) {
-  const raw = web3.utils.hexToBytes(signature);
+  // ethers.Signature.from(signature) throw with non-canonical s or high v, so we do the parsing manually
+  const raw = ethers.getBytes(signature);
   switch (raw.length) {
     case 64:
       return [
-        web3.utils.bytesToHex(raw.slice(0, 32)), // r
-        web3.utils.bytesToHex(raw.slice(32, 64)), // vs
+        ethers.hexlify(raw.slice(0, 32)), // r
+        ethers.hexlify(raw.slice(32, 64)), // vs
       ];
     case 65:
       return [
         raw[64], // v
-        web3.utils.bytesToHex(raw.slice(0, 32)), // r
-        web3.utils.bytesToHex(raw.slice(32, 64)), // s
+        ethers.hexlify(raw.slice(0, 32)), // r
+        ethers.hexlify(raw.slice(32, 64)), // s
       ];
     default:
       expect.fail('Invalid signature length, cannot split');
   }
 }
```

Comment:
We can remove it almost everywhere using `toSignature` instead except for the `reverts with high-s value signature` tests since it's impossible to instantiate a high-s signature in ethers. We can get `r, s, v` values in the following way:

```js
const r = ethers.dataSlice(highSSignature, 0, 32);
const s = ethers.dataSlice(highSSignature, 32, 64);
const v = ethers.dataSlice(highSSignature, 64, 65);
```